### ISSUE TITLE
Any plans for Windows binaries?

### ISSUE THREAD

**Issue by andybak** [NVlabs/instant-ngp#11]
Probably my C++  naivety but I'm struggling to get the makefile to work. A binary release would be a godsend.

**Comment by mpottinger** [comment]
Not exactly what you're asking for but I've had luck in WSL2, works at near native speeds, only 1 FPS difference roughly from natively booted Ubuntu. It is usually easier to compile these kinds of things on Linux in my experience.

**Comment by mpottinger** [comment]
Not exactly what you're asking for but I've had luck in WSL2, works at near native speeds, only 1 FPS difference roughly from natively booted Ubuntu. It is usually easier to compile these kinds of things on Linux in my experience.

**Comment by andybak** [comment]
> It is usually easier to compile these kinds of things on Linux in my experience.

Agreed but configuring CUDA with GPU support under WSL isn't exactly a cakewalk. Are you on Windows 10 or 11? Insider ring or standard?

**Comment by Tom94** [comment]
I've got it running on regular Windows 11 (non-insider), where it compiled without GUI support out of the box.

To get a functioning GUI, you also need to install the latest WSL CUDA driver from https://developer.nvidia.com/cuda/wsl

**Comment by Tom94** [comment]
Regarding a binary release: the codebase unfortunately needs to perform a number of compile-time decisions for good performance on the latest GPUs. This means that a universal binary that runs everyone would be slower, even for people with the highest-end setup.

The only reasonable option I can think of would be to release a number of binaries for different GPU architectures, say
- __86__ (RTX 30XX)
- __75__ (RTX 20XX)
- __61__ (GTX 10XX)
- and __37__ for anything earlier

Since the codebase is still so much in flux, I'm not assigning this a super high priority. But I'm hopeful this will be automatable and doable within GitHub workflows, which already run these builds for CI purposes.

**Comment by mpottinger** [comment]
> > It is usually easier to compile these kinds of things on Linux in my experience.
> 
> Agreed but configuring CUDA with GPU support under WSL isn't exactly a cakewalk. Are you on Windows 10 or 11? Insider ring or standard?

Windows 11, standard I believe. It's not so bad as long as you're familiar with Ubuntu. You must not install the standard cuda drivers in Ubuntu, that is a mistake that I made and it completely messes things up, wsl2 needs to be completely uninstalled and reinstalled in that case. 

Just follow the directions on both the Microsoft site and Nvidia. Install the special NVidia graphics drivers first, and install from the wsl2 cuda repo in ubuntu

[https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl](https://docs.microsoft.com/en-us/windows/ai/directml/gpu-cuda-in-wsl)
